lib/phlex/context.rb

```diff
@@ -26,12 +26,14 @@ def component(component, *args, **kwargs, &block)
       self << component.new(*args, **kwargs, &block)
     end
 
-    def _template_tag(...)
-      _standard_element("template", ...)
+    def _template_tag(*args, **kwargs, &)
+      _standard_element(*args, _name: "template", **kwargs, &)
     end
 
-    def _standard_element(name, content = nil, **kwargs, &block)
+    def _standard_element(content = nil, _name: nil, **kwargs, &block)
       raise ArgumentError if content && block_given?
+
+      name = _name ||= __callee__.name
```

Comment:
Do we need this `name` variable here? Maybe `_name ||= __callee__.name` would be sufficient what do you think?

Reply:
No need for it, I just think we may as well since we then don't need to use the underscored variable name later. AFAIK local variable assignment is optimised away anyway so this shouldn't have a performance impact.

Reply:
> AFAIK local variable assignment is optimised away anyway 

No it's not. Local variables can be inspected through `binding` etc, so Ruby can't eliminate them.

They don't have a big cost though, one more local variable won't make a noticeable difference.